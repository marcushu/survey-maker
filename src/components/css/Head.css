#rowMain {
    background-color: rgb(1, 85, 134);
}

#headMain {
    display: flex;
    justify-content: space-between;
}

#headTitle {
    color: white;
    font-size: 36px;
    padding: 5px;
}

button.goto {
    color:rgb(1, 85, 134);    
    background-color:  white;
    border-style: none;   
    font-size: 18px;
    font-weight: bold;
    margin-top: 17px;
}

button:hover {
    color: rgb(81, 174, 236);
}