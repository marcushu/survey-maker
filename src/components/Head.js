import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import PropTypes from 'prop-types';
import './css/Head.css';


export default function Head({ goto, buttonLabel }) {
  return (
    <Row id="rowMain">
      <Col id="headMain">
        <div id="headTitle">
          Survey Maker
        </div>
        <div>
          <button className="goto" onClick={goto}>
            {buttonLabel}
          </button>
        </div>
      </Col>
    </Row>
  )
}

Head.propTypes = {
  goto: PropTypes.func,
  buttonLabel: PropTypes.string
}