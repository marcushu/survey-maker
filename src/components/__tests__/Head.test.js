import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Head from '../Head';

describe("header component Head", () => {
    const mockFunc = jest.fn();

    it("renders", () => {
        const { asFragment } = render(
            <Head
                goto={mockFunc}
                buttonLabel="Mock Button Label" />)

        expect(asFragment()).toMatchSnapshot();
    });

    it("correctly renders the button", () => {
        const { getByRole } = render(
            <Head
                goto={mockFunc}
                buttonLabel="Mock Button Label" />);

        expect(getByRole('button', { name: /Mock Button Label/i })).toBeInTheDocument();
    });

    it("calls the prop function when clicked", () => {
        const { getByRole } = render(
            <Head
                goto={mockFunc}
                buttonLabel="Mock Button Label" />);
                
        fireEvent.click(getByRole('button', { name: /Mock Button Label/i }));

        expect(mockFunc).toHaveBeenCalledTimes(1);
    })
})