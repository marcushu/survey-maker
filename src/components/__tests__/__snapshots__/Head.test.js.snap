// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component Head renders 1`] = `
<DocumentFragment>
  <div
    class="row"
    id="rowMain"
  >
    <div
      class="col"
      id="headMain"
    >
      <div
        id="headTitle"
      >
        Survey Maker
      </div>
      <div>
        <button
          class="goto"
        >
          Mock Button Label
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
